src/ServiceControl/ExternalIntegrations/FailedMessagesUnArchivedPublisher.cs

```diff
@@ -0,0 +1,35 @@
+namespace ServiceControl.ExternalIntegrations
+{
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Threading.Tasks;
+    using Contracts.MessageFailures;
+    using Raven.Client;
+
+    class FailedMessagesUnArchivedPublisher : EventPublisher<FailedMessagesUnArchived, FailedMessagesUnArchivedPublisher.DispatchContext>
+    {
+        protected override DispatchContext CreateDispatchRequest(FailedMessagesUnArchived @event)
+        {
+            return new DispatchContext
+            {
+                MessageIds = @event.DocumentIds,
+                MessagesCount = @event.MessagesCount
+            };
+        }
+
+        protected override Task<IEnumerable<object>> PublishEvents(IEnumerable<DispatchContext> contexts, IAsyncDocumentSession session)
+        {
+            return Task.FromResult(contexts.Select(r => (object)new Contracts.FailedMessagesUnArchived
+            {
+                MessagesCount = r.MessagesCount,
+                DocumentIds = r.MessageIds
```

Comment:
Why are we publishing DocumentIds? How is that relevant for someone observing this event that is external to RavenDB? Shouldn't this be called MessagesIds?

Reply:
Make sense, I'll change the contract